examples/material_gallery/lib/demo/fitness_demo.dart

```diff
@@ -0,0 +1,486 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:math' as math;
+import 'dart:ui' as ui;
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_sprites/flutter_sprites.dart';
+
+ImageMap _images;
+SpriteSheet _sprites;
+
+class FitnessDemo extends StatefulComponent {
+  FitnessDemo({ Key key }) : super(key: key);
+  _FitnessDemoState createState() => new _FitnessDemoState();
+}
+
+class _FitnessDemoState extends State<FitnessDemo> {
+
+  Future _loadAssets(AssetBundle bundle) async {
+    _images = new ImageMap(bundle);
+    await _images.load(<String>[
+      'packages/flutter_gallery_assets/jumpingjack.png',
+    ]);
+
+    String json = await DefaultAssetBundle.of(context).loadString('packages/flutter_gallery_assets/jumpingjack.json');
+    _sprites = new SpriteSheet(_images['packages/flutter_gallery_assets/jumpingjack.png'], json);
+  }
+
+  void initState() {
+    super.initState();
+
+    AssetBundle bundle = DefaultAssetBundle.of(context);
+    _loadAssets(bundle).then((_) {
+      setState(() {
+        assetsLoaded = true;
+        workoutAnimation = new _WorkoutAnimationNode(onValueUpdate: (){
+          setState((){});
+        });
+      });
+    });
+  }
+
+  bool assetsLoaded = false;
+  _WorkoutAnimationNode workoutAnimation;
+
+  Widget build(BuildContext context) {
+    if (!assetsLoaded) {
+      return new Scaffold(
+        toolBar: new ToolBar(
+          center: new Text("Fitness")
+        )
+      );
+    }
+
+    return new Scaffold(
+      toolBar: new ToolBar(
+        center: new Text("Fitness")
+      ),
+      body: new Material(
+        child: new Column(
+          justifyContent: FlexJustifyContent.center,
+          children: <Widget>[
+            _createWorkoutAnimation(),
```

Comment:
Please inline builder functions into the build() method so it's more obvious how big the build is.
As a general rule, we should only factor out subparts of build functions if they are repeated.
Even better, though, as noted above, would be to factor some of these parts out into separate stateless components.